packages/taraxa-ui/src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react';
+import { InputAdornment, InputBase } from '@material-ui/core';
+import useStyles from './SearchInput.styles';
+import { Search } from '../Icons';
+
+export interface SearchInputProps {
+  placeholder?: string;
+  fullWidth?: boolean;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => any;
+  className?: string;
+}
+
+const SearchInput = ({ placeholder, fullWidth = false, className, onChange }: SearchInputProps) => {
+  const classes = useStyles();
+  const [value, setValue] = useState('');
+
+  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.preventDefault();
+    setValue(e.target.value);
+    if (typeof onChange === 'function') onChange(e);
+  };
+  return (
+    <InputBase
```

Comment:
It should be somewhat like the MUI Autocomplete: https://mui.com/material-ui/react-autocomplete/

Reply:
Depends on how you wanna implement it, the partial search results can go either in a separate, global state, or in here, but it will be a spaghetti component then...

Reply:
SearchInput is a reusable component as you asked @Malak67. You can check the storybook to have an idea on how it works.

Reply:
Also @VargaElod23 , remember we are using MUI v4. (This was the version Leo started it. If we decided to upgrade it we will need to do changes I think)